source/common/http/filter_manager.h

```diff
@@ -1064,7 +1064,9 @@ class DownstreamFilterManager : public FilterManager {
                      parent_filter_state, filter_state_life_span),
         local_reply_(local_reply),
         avoid_reentrant_filter_invocation_during_local_reply_(Runtime::runtimeFeatureEnabled(
-            "envoy.reloadable_features.http_filter_avoid_reentrant_local_reply")) {}
+            "envoy.reloadable_features.http_filter_avoid_reentrant_local_reply")) {
+    stream_info_.setFilterChainName(connection.streamInfo().filterChainName());
```

Comment:
Fix looks good, but I'm wondering how we get better at structurally propagating connection properties to stream info and making them available in situations such as access logs, since I feel this might arise for other existing or future connection properties. Thoughts?

Reply:
Reasonable requirement and I had thought it. There is a shared `ConnectionInfoProvider` that used to provide some connection info. This provider will be set in both connection stream info and request stream info.

I think we can extend the `ConnectionInfoProvider` to set/get filter chain name. wdyt?

Reply:
It seems a bit weird, since this is a config artifact rather than connection property. That said, AFAICT, the network file chain selection is a only depending on connection properties (as per https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/listener/v3/listener_components.proto#envoy-v3-api-msg-config-listener-v3-filterchainmatch), so it could work.

Reply:
Or We can abstract a simlilar new `FilterChainInfoProvider` to supply these filter chain level config artifact? Although I think this is a little heavy because we only have filter chain name that need to be set for now.

Reply:
Maybe try putting it in ConnectionInfoProvider? In theory could we just rename that to ListenerInfoProvideR?

Reply:
Sorry, but I am not sure I followed it correctly.

If you mean put the new `ListenerInfoProvider/FilterChainInfoProvider` in the the ConnectionInfoProvider, then I think it just OK because it's possible to reference the parent litener's info in the connection info.

But if you mean put the filter chain name into the ConnectionInfoProvider and rename the ConnetionInfoProvider to the ListenerInfoProvider, I think it's weird because most methods of current `ConnetionInfoProvider` is used to get connection properties.

Reply:
OK, just do the former then, makes sense ( put the new ListenerInfoProvider/FilterChainInfoProvider in the the ConnectionInfoProvider)

Reply:
Got it,will try to do that these days.